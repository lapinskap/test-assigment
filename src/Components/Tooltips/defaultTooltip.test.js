import React from 'react';
import renderer from 'react-test-renderer';
import DefaultTooltip from './defaultTooltip';
import '../../tests/setupTests';

describe('<DefaultTooltip />', () => {
  const setState = jest.fn();
  const useStateSpy = jest.spyOn(React, 'useState');
  useStateSpy.mockImplementation((init) => [init, setState]);

  let type; let content; const placement = 'auto'; let
    id;

  beforeEach(() => {
    type = 'info';
    content = [];
    id = 'TooltipDemo';
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('renders correctly with props', () => {
    const div = document.createElement('div');
    div.setAttribute('id', 'TooltipDemo');
    document.body.appendChild(div);

    const tree = renderer
      .create(
        <DefaultTooltip
          type={type}
          content={content}
          placement={placement}
          id={id}
        />, div,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with props 2', () => {
    const div = document.createElement('div');
    div.setAttribute('id', 'TooltipDemo');
    document.body.appendChild(div);

    const tree = renderer
      .create(
        <DefaultTooltip
          type="legacy"
          content={content}
          placement={placement}
          id={id}
        />, div,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
