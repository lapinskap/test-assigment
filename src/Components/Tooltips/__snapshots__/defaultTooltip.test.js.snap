// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DefaultTooltip /> renders correctly with props 1`] = `
<i
  className="pe-7s-help1 text-primary font-weight-bold"
  id="TooltipDemo"
/>
`;

exports[`<DefaultTooltip /> renders correctly with props 2 1`] = `
<i
  className="pe-7s-help1 text-danger font-weight-bold"
  id="TooltipDemo"
/>
`;
